lib/asset/AssetGenerator.js

```diff
@@ -335,12 +337,24 @@ class AssetGenerator extends Generator {
 							);
 						assetInfo = mergeAssetInfo(assetInfo, info);
 						assetPath = JSON.stringify(path + filename);
+						assetPathForCss = path + filename;
 					} else {
 						runtimeRequirements.add(RuntimeGlobals.publicPath); // add __webpack_require__.p
 						assetPath = runtimeTemplate.concatenation(
 							{ expr: RuntimeGlobals.publicPath },
 							filename
 						);
+						const compilation = runtimeTemplate.compilation;
+						const path =
+							compilation.outputOptions.publicPath === "auto"
+								? CssUrlDependency.PUBLIC_PATH_AUTO
+								: compilation.getAssetPath(
+										compilation.outputOptions.publicPath,
+										{
+											hash: compilation.hash
+										}
+									);
+						assetPathForCss = path + filename;
```

Comment:
Maybe we can generate this only when asset module in CSS dependecy to avoid extra work, because we still need to improve it, have
`runtimeRequirements.add(RuntimeGlobals.publicPath);`, and so we generate extra JS code if you have an asset in your CSS (but not in JS), and in future we will need to fix it too (I described it above)

Reply:
Yes, both need the asset module to aware whether it is only referenced by a css module or a js module, since it's not very related to this PR, I can give a try in the future PR